spec/ios_l10n_helper_spec.rb

```diff
@@ -0,0 +1,162 @@
+    end
+
+    it 'returns nil on file that is not a strings file at all' do
+      png_fixture = fixture('not-a-strings-file.png')
+      expect(File).to exist(png_fixture)
+      expect(described_class.strings_file_type(path: '/invalid-path')).to be_nil
+    end
+  end
+
+  describe '#merge_strings' do
+    # Ensuring we test this against input files which are using different encodings (UTF-16BE & UTF-8) is an important case to cover
+    it 'properly merges 2 textual strings files with different encodings into a new one' do
+      paths = [fixture('Localizable-utf16.strings'), fixture('InfoPlist-utf8.strings')]
+
+      # Just making sure we don't accidentally change the encoding of the fixture files in the future
+      expect(file_encoding(paths[0])).to eq(Encoding::UTF_16BE)
+      expect(file_encoding(paths[1])).to eq(Encoding::UTF_8)
+
+      Dir.mktmpdir('a8c-release-toolkit-l10n-helper-tests-') do |tmp_dir|
+        output_file = File.join(tmp_dir, 'output.strings')
+        described_class.merge_strings(paths: paths, output_path: output_file)
+        expect(File.read(output_file)).to eq(File.read(fixture('expected-merged.strings')))
+      end
+    end
+
+    it 'properly merges 2 textual strings files in-place into the first one' do
+      # This test is especially useful to check that if we use one of the input file as output, things will still work.
+      paths = [fixture('Localizable-utf16.strings'), fixture('InfoPlist-utf8.strings')]
+      Dir.mktmpdir('a8c-release-toolkit-l10n-helper-tests-') do |tmp_dir|
+        paths.each { |f| FileUtils.cp(f, tmp_dir) }
+        paths.map! { |f| File.join(tmp_dir, File.basename(f)) }
+        described_class.merge_strings(paths: paths, output_path: paths.first)
+        expect(File.read(paths.first)).to eq(File.read(fixture('expected-merged.strings')))
+      end
+    end
+
+    it 'returns duplicate keys found' do
+      paths = [fixture('Localizable-utf16.strings'), fixture('non-latin-utf16.strings')]
+      Dir.mktmpdir('a8c-release-toolkit-l10n-helper-tests-') do |tmp_dir|
+        output_file = File.join(tmp_dir, 'output.strings')
+        duplicates = described_class.merge_strings(paths: paths, output_path: output_file)
+        expect(duplicates).to eq(%w[key1 key2])
+      end
+    end
+
+    it 'raises if one of the strings file is not in textual format' do
+      paths = [fixture('Localizable-utf16.strings'), fixture('xml-format.strings')]
+      Dir.mktmpdir('a8c-release-toolkit-l10n-helper-tests-') do |tmp_dir|
+        output_file = File.join(tmp_dir, 'output.strings')
+        expect do
+          described_class.merge_strings(paths: paths, output_path: output_file)
+        end.to raise_exception(RuntimeError, "The file `#{paths[1]}` is in xml format but we currently only support merging `.strings` files in text format.")
+      end
+    end
+
+    it 'raises if one of the strings file does not exist' do
```

Comment:
Nitpick:
```suggestion
    it 'raises if any of the strings file does not exist' do
```
Reading "one" made me wonder "what happens if more than one doesn't exist?" But... it could be just my contrived mind